### PR TITLE
FINERACT-822 Enforce StringSplitter check

### PR DESCRIPTION
**Pull request by percyashu**
FINERACT-822

### CODE REVIEW

**Comment by vorburger** [comment]
@awasum will you be doing a more detailed review of this PR? (Your GSoC student.)

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/ColumnValidator.java [hunk 1]
```diff
@@ -122,6 +122,7 @@ public void validateSqlInjection(String schema, String... conditions) {
         return tableColumnMap;
     }
 
+    @SuppressWarnings("StringSplitter")
```
Comment (awasum) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/ColumnValidator.java:125]: Why did you suppress this?

**Review by awasum — APPROVED** [review]
This follows the advice as seen here:https://errorprone.info/bugpattern/StringSplitter

The key thing is just looking the way the lists are accessed so make sure they correspond to the index of the arrays been refactored. I think this can be merged except am missing something... atleast the tests have passed.

@percyashu one little thing which was mentioned in the link above:
'`TIP: if you use Splitter, consider extracting the instance to a static final field.`

**Comment by awasum** [comment]
@vorburger merge this when you can. Or maybe you want to take another quick look to make sure am not missing something. @ptuomola @xurror @thesmallstar  take a look.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/ColumnValidator.java (hunk 1 above)
Comment (percyashu) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/ColumnValidator.java:125]: Using the Splitter implementation in this method causes an errors in some integration test, I looked into but wasnt making any inroads to find out what was causing this. Will create an issue on issue tracker to remove the annotation.

**Pull request merged**